Sources/Confidence/ConfidenceEventSender.swift

```diff
@@ -3,7 +3,7 @@ import Foundation
 /**
 Sends events to Confidence. Contextual data is appended to each event
 */
-public protocol ConfidenceEventSender: Contextual {
+public protocol ConfidenceEventSender: ConfidenceContextProvider {
```

Comment:
Would removing Contextual not be a breaking change? I am not sure whether it's necessary to remove 🤔

Reply:
Answering here for @vahidlazio as well: the problem I found with `Contextual` is that [withContext returns a Self](https://github.com/spotify/confidence-sdk-swift/blob/82d69a0f7087e7147de87071bf5cf52e9bfbfc3d/Sources/Confidence/Contextual.swift#L21), and if it's `Confidence` implementing such protocol the return type can only be `Confidence` (while we want `ConfidenceEventSender` instead)

Reply:
I don't _think_ that removing `Contextual` is a breaking change, since final users never operate with such type directly: they either have a Confidence instance or a ConfidenceEventSender instance

Reply:
> I think my issue with this is that removing Contextual then means this structure doesn't align with Android/JS anymore.

True, I am also not very about that. But as long as this is internals, it's less of a problem and we can always change it without breaking backwards compat with final users

> Right, so since Confidence doesn't confirm to EventSender, it won't be able to return one when calling withContext.

Even if it would conform to EventSender, it seems like the return type `Self` only accepts the type of the implementing class `Confidence`